spark-jobs/src/main/scala/za/co/absa/enceladus/standardization/config/StandardizationConfigParser.scala

```diff
@@ -137,64 +148,66 @@ object StandardizationConfigParser {
 
-      opt[String]("null-value").optional()
-        .action((value, config) => config.withFixedWidthNullValue(Some(value)))
-        .text("""For FixedWidth file format. Sets the representation of a null value. Defaults is ""."""),
-
       checkConfig(checkConfigX(_, builder))
     )
   }
   //scalastyle:on method.length
 
   private val formatsSupportingCharset = List("xml", "csv", "json", "cobol", "fixed-width")
 
-  private def typicalError(field: String, format: String): String = {
-    s"The $field option is supported only for $format format"
+  private def unsupportedOptionError(option: String, format: String): String = {
+    unsupportedOptionError(option, Seq(format))
+  }
+
+  private def unsupportedOptionError(option: String, formats: Seq[String]): String = {
+    def sprintf(frmt: String, s: String = ""): String = s"The $option option is supported only for $frmt format$s"
+
+    formats match {
+      case Seq()       => ""
+      case Seq(format) => sprintf(format)
+      case _           =>
+        val revertedFormats = formats.reverse
+        val frmt = revertedFormats.reverse.mkString(", ") + " and " + revertedFormats.head
+        sprintf(frmt, "s")
+    }
   }
 
   private def checkCharset[R <: StandardizationConfigParser[R]](config: R): List[String] = {
     if (!formatsSupportingCharset.contains(config.rawFormat) && config.charset.isDefined) {
-      List(typicalError("--charset", "CSV, JSON, XML, COBOL and FixedWidth"))
+      List(unsupportedOptionError("--charset", Seq(csvFormatName, jsonFormatName, xmlFormatName, cobolFormatName, fixedWidhFormatName)))
     } else {
       List.empty
     }
   }
 
   private def checkXMLFields[R <: StandardizationConfigParser[R]](config: R): List[String] = {
     if (config.rowTag.isDefined && config.rawFormat != "xml") {
-      List(typicalError("--row-tag", "XML raw data"))
+      List(unsupportedOptionError("--row-tag", xmlFormatName))
     } else {
       List.empty
     }
   }
 
-  private def checkCSVFields[R <: StandardizationConfigParser[R]](config: R): List[String] = {
-    def csvFieldsThatShouldNotBePresent(config: R): List[String] = {
-      val format = "CSV"
-      val definedFields = Map(
-        typicalError("--delimiter", format) -> config.csvDelimiter.isDefined,
-        typicalError("--escape", format) -> config.csvEscape.isDefined,
-        typicalError("--header", s"$format raw data") -> config.csvHeader.contains(true),
-        typicalError("--quote", format) -> config.csvQuote.isDefined
-      )
-      definedFields.filter { case (_, value) => value }.keys.toList
-    }
-
+  private def checkCSVFields[R <: StandardizationConfigParser[R]](config: R): Seq[String] = {
     if (config.rawFormat == "csv") {
-      List.empty
+      Seq.empty
     } else {
-      csvFieldsThatShouldNotBePresent(config)
+      Seq (
+        config.csvDelimiter.map(_ => unsupportedOptionError("--delimiter", csvFormatName)),
```

Comment:
This seams to me is better readable/understandable, then the original function with a sub-function.
Please comment and feel free to disagree 😉